### PR TITLE
Feat/comment old node version

### PR DESCRIPTION
**Pull request by m-kusnierz**
[Example](https://github.com/sumup/invoices-merchant-sync/pull/367) from service with Node 18

image

---

[No comments](https://github.com/sumup/accounting-integration-services/pull/282) in service with Node 20

image

### CODE REVIEW

File: npm-install/action.yml [hunk 1]
```diff
@@ -156,3 +156,23 @@ runs:
...
+
+    - if: always() && fromJSON(steps.determine-old-node-version.outputs.outdated-node-version)
+      name: Comment old NodeJs version
+      uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v.4.0.0
+      with:
+        issue-number: ${{ github.event.number }}
+        edit-mode: replace
+        body: |
+          The node version specified in `package.json` is ${{ steps.determine-node-npm-version.outputs.node-version }}. 
+          Please update it. Don't forget about Dockerfile update if this is a service
```
Comment (m-kusnierz) [inline npm-install/action.yml:178]: @hilleer Feel free to reword this comment 🙂

File: npm-install/action.yml [hunk 2]
```diff
@@ -156,3 +156,23 @@ runs:
...
+
+    # Comment old NodeJs version
+    - if: always()
+      id: determine-old-node-version
+      name: Determine old NodeJs version
+      shell: bash
+      env:
+        NODE_VERSION: ${{ steps.determine-node-npm-version.outputs.node-version }}
+      # bash can only compare integers, use 'bc' to compare versions like 18.20. Also remove '.x' wildcards that we commonly use in version
+      run: echo outdated-node-version=$(echo "${NODE_VERSION//.x/} < 20" | bc) >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (hilleer) [reply npm-install/action.yml:168]: Good question. We may ask the team next week. We can start with node 18 for now and change to node 20 some time at the end of this year or the beginning of next? Node 18 has support until April.

File: npm-install/action.yml [hunk 3]
```diff
@@ -156,3 +156,23 @@ runs:
...
           Please commit updated `package-lock.json` file (otherwise deployment will fail).
+
+    # Comment old NodeJs version
+    - if: always()
+      id: determine-old-node-version
+      name: Determine old NodeJs version
+      shell: bash
+      env:
+        NODE_VERSION: ${{ steps.determine-node-npm-version.outputs.node-version }}
+      # bash can only compare integers, use 'bc' to compare versions like 18.20. Also remove '.x' wildcards that we commonly use in version
```
Comment (hilleer) [inline npm-install/action.yml:167]: Nice comment

File: npm-install/action.yml (hunk 1 above)
Comment (hilleer) [inline npm-install/action.yml:178]: "the node version is" - what is it? 😄 

```suggestion
          The node version "${{ steps.determine-node-npm-version.outputs.node-version }}" specified in `package.json` is <placeholder - insert better text about dropped LTS support> and should be bumped to v20 minimum. 
          Note: Remember to update node version in Dockerfile too.
```

(docker part I guess we can see from the template used if it should be commented or not? Or is that not possible?

**Review by hilleer — APPROVED** [review]
Super nice!!! 🚀

File: npm-install/action.yml (hunk 1 above)
Comment (hilleer) [reply npm-install/action.yml:178]: Could also include a link to this page maybe? https://nodejs.org/en/about/previous-releases

File: npm-install/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-kusnierz) [reply npm-install/action.yml:168]:
```suggestion
      run: echo outdated-node-version=$(echo "${NODE_VERSION//.x/} < 18" | bc) >> $GITHUB_OUTPUT
```

File: npm-install/action.yml (hunk 1 above)
Comment (m-kusnierz) [reply npm-install/action.yml]: > docker part I guess we can see from the template used if it should be commented or not? Or is that not possible?

We don't know whether this is a service or a package here 😞

File: npm-install/action.yml [hunk 4]
```diff
@@ -156,3 +156,23 @@ runs:
...
+      run: echo outdated-node-version=$(echo "${NODE_VERSION//.x/} < 18" | bc) >> $GITHUB_OUTPUT
+
+    - if: always() && fromJSON(steps.determine-old-node-version.outputs.outdated-node-version)
+      name: Comment old NodeJs version
+      uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v.4.0.0
+      with:
+        issue-number: ${{ github.event.number }}
+        edit-mode: replace
+        body: |
+          The node version "${{ steps.determine-node-npm-version.outputs.node-version }}" specified in `package.json` is <placeholder - insert better text about dropped LTS support> and should be bumped to v20 minimum.
```
Comment (m-kusnierz) [inline npm-install/action.yml:177]: How about this @hilleer?
```suggestion
          The node version "${{ steps.determine-node-npm-version.outputs.node-version }}" specified in `package.json` is not supported anymore or close to end of support and should be bumped to v18 minimum. 
          Check the Node.js releases page for details https://nodejs.org/en/about/previous-releases
```

Comment (hilleer) [reply npm-install/action.yml:177]: Sure. Or:

```suggestion
          The node version "${{ steps.determine-node-npm-version.outputs.node-version }}" specified in `package.json` is not supported anymore or close to end of support and should be bumped to `>= 18`. 
```

Comment (m-kusnierz) [reply npm-install/action.yml]: 👍 `>= 18` will be easier to spot in the comment

File: npm-install/action.yml [hunk 5]
```diff
@@ -156,3 +156,23 @@ runs:
...
+      run: echo outdated-node-version=$(echo "${NODE_VERSION//.x/} < 18" | bc) >> $GITHUB_OUTPUT
+
+    - if: always() && fromJSON(steps.determine-old-node-version.outputs.outdated-node-version)
+      name: Comment old NodeJs version
+      uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v.4.0.0
+      with:
+        issue-number: ${{ github.event.number }}
+        edit-mode: replace
+        body: |
+          The node version "${{ steps.determine-node-npm-version.outputs.node-version }}" specified in `package.json` is not supported anymore or close to end of support and should be bumped to `>= 18`.
```
Comment (m-kusnierz) [inline npm-install/action.yml:177]:
```suggestion
          The node version `${{ steps.determine-node-npm-version.outputs.node-version }}` specified in `package.json` is not supported anymore or close to end of support and should be bumped to `>= 18`. 
```

**Comment by m-kusnierz** [comment]
I'm merging this. 

Here's how it looks in the PR.
image

 I wonder if there is a way to tag the code owners without reading the CODEOWNERS file 🤔

**Pull request merged**